# Steps for Camelina seed Time Course Experiment
## Remove Bad Samples from Count Data
```
Rscript --vanilla /home/grabowsky/tools/workflows/rnaseq_TC_contrasts/r_scripts/remove_bad_samp_counts.r
```
## DESeq2
### Overview
#### My Notes
Uses negative binomial generalized linear models and estimated dispersion and log-fold-changes for different parameters in the model
#### Resources
##### Bioconductor Outline
I followed the outline provided by Bioconductor:
http://master.bioconductor.org/packages/release/workflows/vignettes/rnaseqGene/inst/doc/rnaseqGene.html
##### Package Website
https://bioconductor.org/packages/release/bioc/html/DESeq2.html
##### Manual
https://bioconductor.org/packages/release/bioc/manuals/DESeq2/man/DESeq2.pdf
#### Explanation of Files Generated by Scripts
Returns 2 groups of 3 files:
##### Groups
* `<general>`: genes that are different between the Control and Treatment after accounting for the other parameters, specifically, time. These genes "overall" differences and not necessarily different responses/reactions across time. Usually, this will be a large list.
* `<TC>`: genes that show differences at different time points that are different from the "global" differences between Control and Treatment. For instance: Expression in Treatment line increases from T0 to T1 but is unchanged in Control line.
##### Files Within Groups
* `<PREFIX_DESeq2_GROUP_genes.txt>`: contains all genes, overall p-value, and multiple testing-corrected p-value
* `<PREFIX_DESeq2_TC_var_lfc.txt>`: (only for TC group) contains the estimated log-fold-change associated with each variable in the model (not including TimeVsTime variables). This can be used for trying to cluster genes based on their modeled response across conditions
* `<PREFIX_DESeq2_GROUP_full_mat.txt>`: contains the gene name, average expression parameter values, average expression, F value, overall p-value, and multiple testing-adjusted p-value for the significant genes
### Identify Differentially Expressed (DE) genes
R script uses two approaches in DESeq2 (general and TC) to identify differentially expressed genes.
#### R Script
```
/home/grabowsky/tools/workflows/rnaseq_TC_contrasts/r_scripts/DESeq2_TC_script.r
```
#### Shell Script
```
/home/grabowsky/tools/workflows/rnaseq_TC_contrasts/sh_scripts/run_DESeq2_TC_2conditions.sh
```
#### Run on Full Data Set
```
qsub -cwd -N cam_splinetime -l h_vmem=4G -q all.q /home/grabowsky/tools/workflows/rnaseq_TC_contrasts/sh_scripts/run_DESeq2_TC_2conditions.sh
```
### Output files
#### File Directory
`/home/t4c1/WORK/grabowsk/data/Camelina_suneson_seed_TC`
#### General DE
Any difference between MT5 and 167; MANY genes
* `Camelina_TC_DESeq2_general_genes.txt`
  * Gene names and p-values
* `Camelina_TC_DESeq2_general_full_mat.txt`
  * Gene names, p-values, and a few other values
## maSigPro
### Overview
#### My Notes
Fits regression including different parameters for relationship with time (ex: linear, quadratic), then tests how different the fit is when parameters are modeled separately for the treatment vs modeled for treatment and control together. Then goes back and removed parameters to test which parameters are important for the difference.
#### Resources
##### maSigPro Website
https://www.bioconductor.org/packages/release/bioc/html/maSigPro.html
##### maSigPro Bioconductor Users Guide
https://www.bioconductor.org/packages/release/bioc/vignettes/maSigPro/inst/doc/maSigProUsersGuide.pdf
##### maSigPro Reference Manual
https://www.bioconductor.org/packages/release/bioc/manuals/maSigPro/man/maSigPro.pdf
#### Explanation of Files Generated by Scripts
Returns 3 files:
* `<PREFIX_DE_genes.txt>`: contains all genes and overall p-value (corresponding to the difference across all parameters)
* `<PREFIX_DE_betas.txt>`: contains the gene names and the betas for all the parameters, for the significant genes. This can be used for trying to cluster genes based on their modeled response across conditions
* `<PREFIX_DE_full_mat.txt>`: contains the gene name, overall p-value, R^2, and beta and p-value for each parameter for the significant genes
### Identify DE Genes
#### R Script
```
/home/grabowsky/tools/workflows/rnaseq_TC_contrasts/r_scripts/maSigPro_TC_script.r
```
#### Shell Script
```
/home/grabowsky/tools/workflows/rnaseq_TC_contrasts/sh_scripts/run_maSigPro.sh
```
#### Run Script
```
qsub -cwd -N cam_masigpro -l h_vmem=4G -q all.q /home/grabowsky/tools/workflows/rnaseq_TC_contrasts/sh_scripts/run_maSigPro.sh
```
## splineTimeR
### Overview
#### My Notes
Fits spline parameters for how expression reponds at different time points
#### Resources
##### Bioconductor Website
https://bioconductor.org/packages/release/bioc/html/splineTimeR.html
##### Users Manual
https://bioconductor.org/packages/release/bioc/vignettes/splineTimeR/inst/doc/splineTimeR.pdf
##### Reference Manual
https://bioconductor.org/packages/release/bioc/manuals/splineTimeR/man/splineTimeR.pdf
#### Explanation of Files Generated by Script
Returns two groups of three files
##### Groups
* `<noIntercept>`: genes with statistically different spline parameters between Treatment and Control; does not include genes that are DE across all times at a similar level, only those with different responses through time
* `<withIntercept>`: genes from `<noIntercept>` as well as those that have consistent DE across all times
##### Files within Groups
* `<PREFIX_splineTimeR_GROUP_genes.txt>`: contains all genes, overall p-value, and multiple testing-corrected p-value (corresponding to the difference across all parameters)
* `<PREFIX_splineTimeR_GROUP_paramVals.txt>`: contains the gene names and spline parameter values for the significant genes. This can be used for trying to cluster genes based on their modeled response across conditions
* `<PREFIX_splineTimeR_GROUP_full_mat.txt>`: contains the gene name, parameter values, average expression, F value, overall p-value, and multiple testing-adjusted p-value for the significant genes
### Identify DE Genes
#### R script
```
/home/grabowsky/tools/workflows/rnaseq_TC_contrasts/r_scripts/splineTimeR_2conditions_script.r
```
#### Shell script
```
/home/grabowsky/tools/workflows/rnaseq_TC_contrasts/sh_scripts/run_splineTimeR_2conditions.sh
```
#### Run script
```
qsub -cwd -N cam_splinetime -l h_vmem=4G -q all.q /home/grabowsky/tools/workflows/rnaseq_TC_contrasts/sh_scripts/run_splineTimeR_2conditions.sh
```

