# Steps for Camelina seed Time Course Experiment
## Remove Bad Samples from Count Data
```
Rscript --vanilla /home/grabowsky/tools/workflows/rnaseq_TC_contrasts/r_scripts/remove_bad_samp_counts.r
```
## DESeq2
### Overview
#### Resources
##### Bioconductor Outline
I followed the outline provided by Bioconductor:
http://master.bioconductor.org/packages/release/workflows/vignettes/rnaseqGene/inst/doc/rnaseqGene.html
##### Package Website
https://bioconductor.org/packages/release/bioc/html/DESeq2.html
##### Manual
https://bioconductor.org/packages/release/bioc/manuals/DESeq2/man/DESeq2.pdf
#### My Notes
Uses negative binomial generalized linear models and estimated dispersion and log-fold-changes for different parameters in the model
#### Explanation of Files Generated by Scripts
Returns 2 groups of 3 files:
##### Groups
* `<general>`: genes that are different between the Control and Treatment after accounting for the other parameters, specifically, time. These genes "overall" differences and not necessarily different responses/reactions across time. Usually, this will be a large list.
* `<TC>`: genes that show differences at different time points that are different from the "global" differences between Control and Treatment. For instance: Expression in Treatment line increases from T0 to T1 but is unchanged in Control line.
##### Files Within Groups
* `<PREFIX_DESeq2_GROUP_genes.txt>`: contains all genes, overall p-value, and multiple testing-corrected p-value
* `<PREFIX_DESeq2_TC_var_lfc.txt>`: (only for TC group) contains the estimated log-fold-change associated with each variable in the model (not including TimeVsTime variables). This can be used for trying to cluster genes based on their modeled response across conditions
* `<PREFIX_DESeq2_GROUP_full_mat.txt>`: contains the gene name, average expression parameter values, average expression, F value, overall p-value, and multiple testing-adjusted p-value for the significant genes
### Identify Differentially Expressed (DE) genes
R script uses two approaches in DESeq2 (general and TC) to identify differentially expressed genes.
#### R Script
```
/home/grabowsky/tools/workflows/rnaseq_TC_contrasts/r_scripts/DESeq2_TC_script.r
```
#### Shell Script
```
/home/grabowsky/tools/workflows/rnaseq_TC_contrasts/sh_scripts/run_DESeq2_TC_2conditions.sh
```
#### Run on Full Data Set
```
qsub -cwd -N cam_splinetime -l h_vmem=4G -q all.q /home/grabowsky/tools/workflows/rnaseq_TC_contrasts/sh_scripts/run_DESeq2_TC_2conditions.sh
```

